app/lib/network_utils.py

```diff
@@ -652,32 +653,33 @@ async def deregister_statefulset_collators(para_id: str, stateful_set_name: str)
     await deregister_collator_nodes(chain, collator_node_names, ss58_format)
 
 
-async def add_invulnerable_collators(para_id, nodes=[], addresses=[]):
-    log.info(f'Adding invulnerables collators to parachain #{para_id}; nodes={nodes} and addresses={addresses}')
-    current_invulnerables = get_collator_selection_invulnerables(para_id)
+async def set_invulnerable_collators(para_id, nodes=[], addresses=[]):
+    log.info(f'Setting invulnerables collators to parachain #{para_id}; nodes={nodes} and addresses={addresses}')
     invulnerables_to_add = []
     for node_name in nodes:
         node_collator_account = get_substrate_node(node_name).get("collator_account")
         if node_collator_account:
             invulnerables_to_add.append(node_collator_account)
```

Comment:
Smelling code.

For example:
1. We call this function with `nodes=['validator-nodee-h']`
2. We may have the following issues: typo in name,  k8s api is down, pod deleted/restarted, we have bug in code. 
2. `node_collator_account` will be `null` or incorrect
3. network set incorrect  `invulnerables` to which we don't have a key, or remove all  `invulnerables`, such issue is hard to debug and may break the chain.

Proposal: Don't use the `set_invulnerable_collators` function in the code (add comment ),  and add a Warning in the API doc, and deprecation warning. 

Note: calling with  `addresses=[...]`, is also dangerous, but it is totally fine since you are controlling the input.

Reply:
Yes I agree this `set_invulnerable_collators` is dangerous, especially with node names with the automatic retrieval of the address. Maybe we should delete this code completely.